[dhlee03] iP

dhlee03 opened the pull request:
### FriendlyTool

> "For every minute spent organising, an hour is earned." – Benjamin Franklin([source](https://www.theplainsimplelife.com/quotes-about-being-organised/))

FriednlyTool helps you to organise tasks easily and save time.

**Why** using it? 

- Easy and Fast to use
- Text-based 
- User-friendly features😉

**Features**:

- [x]  Adding/Managing tasks
- [x]  Viewing a list of tasks
- [x]  Finding tasks

**How** to install?

1. Download it from [here](https://github.com/dhlee03/ip/releases)
2. Open it
3. Add your own task
4. Enjoy its features

And it's _Free_!

If you are a Java programmer, you can use it to practice Java too. Here's the main method:
```java

public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

dhlee03 committed:
initial skeletal version

dhlee03 committed:
adds echo function

dhlee03 committed:
include adding and showing list features

dhlee03 committed:
include ability to mark/unmark task

dhlee03 committed:
added three task types: ToDos, Events, Deadlines

dhlee03 committed:
added Automated Text UI Testing

dhlee03 committed:
includes function for error handling

dhlee03 committed:
added features to delete task

dhlee03 committed:
Use Enums for command types

dhlee03 committed:
added features to save and load task data

dhlee03 committed:
Fixed errors. Added features to make folders if directory is not available.

dhlee03 committed:
Merge branch 'branch-Level-7'

dhlee03 committed:
Support date formats

yashpola commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,283 @@
+
 public class Duke {
+    private boolean isActive;
+    private List<Task> myList;
```
This might be better if it was `tasks` as its more descriptive of what the list is

yashpola commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,283 @@
+        Duke ft = new Duke();
+        ft.init();
+    }
+    public void init() {
```
A more descriptive name might be `greetUser()` as its specific to the context of the class and the client knows what action is triggered by calling this method rather than if it was simply `init`

yashpola commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,283 @@
+                + "    ____________________________________________________________" );
+    }
+
+    private void nextAction(String input) throws ftException {
```
While `nextAction` does imply a continuation of interaction with the user, something like `handleUserResponse` might be better as it makes it clear who is taking the next action (the bot in this case) and who it is meant for (the user)

yashpola commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,283 @@
+        }
+    }
+
+    private void addTask(StringTokenizer st, CommandTypes ct) throws ftException {
```
This codefile is a bit long for me, I think it would benefit from more modularization through perhaps a Parser class or something to handle the parsing of the request and task creation. This also reduces the repeated `switch-case` to check task type before acting on the user request

yashpola commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,283 @@
+        }
+    }
+
+    private void unmark(StringTokenizer st) throws ftException {
```
Since your other methods are in the form of `deleteTask` and `updateTask` etc, to be consistent it would be better if these were `unmarkTask` and `markTask` as well

yashpola reviewed (commented):
The functionality and naming is mostly clear despite the length of the files but for future readability and expansion, my biggest suggestion would be more modularisation most evidently through a class dedicated to string parsing

dhlee03 committed:
Merge branch 'branch-Level-8'

dhlee03 committed:
Added UI and TaskList

dhlee03 committed:
Added Storage

dhlee03 committed:
Added Parser

dhlee03 committed:
Merge branch 'branch-A-MoreOOP'

dhlee03 committed:
divided into multiple packages

dhlee03 committed:
Merge branch 'add-gradle-support'

dhlee03 committed:
Add JUnit tests

dhlee03 committed:
Add JavaDoc

dhlee03 committed:
Merge branch 'branch-A-JavaDoc'

dhlee03 committed:
modify to follow the coding standard

dhlee03 committed:
changed package name

dhlee03 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/friendlytool/command/Parser.java
#	src/main/java/friendlytool/process/Date.java
#	src/main/java/friendlytool/process/Duke.java
#	src/main/java/friendlytool/process/Storage.java
#	src/main/java/friendlytool/process/TaskList.java
#	src/main/java/friendlytool/process/UI.java
#	src/main/java/friendlytool/task/Deadline.java
#	src/main/java/friendlytool/task/Event.java
#	src/main/java/friendlytool/task/Task.java
#	src/main/java/friendlytool/task/ToDo.java

dhlee03 committed:
fix merge error and add javadoc to exception

dhlee03 committed:
Add task find using keyword function

dhlee03 committed:
Add checkstyle with gradle

dhlee03 committed:
Merge branch 'branch-A-CheckStyle'

dhlee03 committed:
Add a GUI to chatbot

dhlee03 committed:
Merge branch 'branch-Level-10'

dhlee03 committed:
Update build gradle file

dhlee03 committed:
Update javadoc based on new changes

GUI requires some functions to return a string rather than
printing those. Hence, there should be some changes to
the original functions' javadoc.

dhlee03 committed:
Add asserts statements

The code has some important assumtions that some of objects
are not null.

The assert statements can help debugging process.

dhlee03 committed:
Merge pull request #2 from dhlee03/branch-A-Assertions

Add asserts statements

dhlee03 committed:
Merge branch 'master' of https://github.com/dhlee03/ip

dhlee03 committed:
Improve code quality

The happy path should be clear and prominent in the code.
The code should not be distracted by alternative paths.

dhlee03 committed:
Merge pull request #3 from dhlee03/branch-A-CodeQuality

Improve code quality

dhlee03 committed:
Merge branch 'master' of https://github.com/dhlee03/ip

dhlee03 committed:
Add sort commands

Sorting deadlines and events chronogically based on the end date.
This helps users to know which task they should do first.

dhlee03 committed:
Update code to store list after sorting

dhlee03 committed:
Merge branch 'branch-BCD-Extension'

dhlee03 committed:
Add background color and text box GUI

dhlee03 committed:
Change class name to FriendlyTool

dhlee03 committed:
Modify bye command to exit the program

dhlee03 committed:
Add a title to the program

dhlee03 committed:
Update README.md

dhlee03 committed:
Update class name Ui

dhlee03 committed:
Rename UI.png to Ui.png

dhlee03 committed:
Merge branch 'master' of https://github.com/dhlee03/ip

dhlee03 committed:
Change path to a png file

dhlee03 committed:
Remove old README file

dhlee03 committed:
Add more Junit tests

dhlee03 committed:
Remove text ui test

dhlee03 committed:
Improve parser methods

dhlee03 committed:
Modify structure of Storage code

dhlee03 committed:
Improve GUI and fix errors
